lib/datadog/core/configuration/option.rb

```diff
@@ -235,8 +255,10 @@ def validate(type, value)
           case type
           when :string
             value.is_a?(String)
-          when :int, :float
-            value.is_a?(Numeric)
+          when :int
+            value.is_a?(Integer) || (value.is_a?(Float) && (value.truncate == value))
+          when :float
+            value.is_a?(Float) || value.is_a?(Integer) || value.is_a?(Rational)
```

Comment:
This makes me think we should change `validate_type` above to coerce int and float to the correct types.

It seems like a bit of a potential sharp edge that someone can set a `Rational` on a `:float` setting, or a `Float` on `:int` but our code may not deal very well with it.

On the other hand, it's nice to allow these; hence why I think here would be a nice place to coerce after validation?